### ISSUE TITLE
Plan view not automatically refreshing via the observe API

### ISSUE THREAD

**Issue by AdamElliott-AB** [nasa/openmct#7129]
Hi @ozyx , I have a custom objectProvider and Plan views are not auto-refreshing as expected. 

Looking at https://github.com/nasa/openmct/blame/6b32c63039e0b3ddc524f65f7304e1342b54427f/src/plugins/plan/components/Plan.vue#L262
, shouldn't the handleSelectFileChange method take the new domainObject as an argument and pass that to the getValidatedData method? I am new to using the observe API so it is possible I have a misunderstanding. 

I did confirm that my objectProvider was being asked to observe the Plan object and was calling the callback when the object changed.

**Comment by AdamElliott-AB** [comment]
No, I am not using CouchDB. I am testing an object provider that I wrote. The latest OpenMCT master was used.
I install my object provider with
`openmct.objects.addProvider('', bucketObjectProvider);`
get, create, and update methods of the objects API are working as expected. I was testing the observe method of the API with the Plan.vue when I noticed the issue.

Can you verify if the Plan view's auto-refresh is working as expected with the current codebase in master and any other object provider such as CouchDB?
I will continue to test my objectProvider with other components besides Plan that use the observe API.

**Comment by akhenry** [comment]
@AdamElliott-AB There is actually a bug here that we only recently discovered ourselves that prevents plans specifically from auto-updating. Stand by and I'll get some more details from the dev who found it.

**Comment by AdamElliott-AB** [comment]
Thanks for the heads up @akhenry . I pulled the latest which didn't fix it for me, but I shortly afterward found my problem. My object update was getting discarded because I did not modify the "persisted" timestamp when I modified the object remotely.
https://github.com/nasa/openmct/blob/6b32c63039e0b3ddc524f65f7304e1342b54427f/src/api/objects/ObjectAPI.js#L667C40-L667C65

Sorry for the noise, and thank you both for your help.
